src/lib/Microsoft.Health.Extensions.Fhir.R4/Service/FhirService.cs

```diff
@@ -0,0 +1,97 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using IFhirClient = Microsoft.Health.Fhir.Client.IFhirClient;
+
+namespace Microsoft.Health.Extensions.Fhir.Service
+{
+    public class FhirService : IFhirService
+    {
+        private readonly IFhirClient _fhirClient;
+
+        public FhirService(IFhirClient fhirClient)
+        {
+            _fhirClient = EnsureArg.IsNotNull(fhirClient, nameof(fhirClient));
+        }
+
+        public async Task<T> CreateResourceAsync<T>(
```

Comment:
It seems like most of these methods are just pass throughs to the same method on the IFhirClient.  Curious as to the overall purpose of the class rather than using the IFhirClient directly?

Reply:
Sure, in a previous comment above we discussed this added layer of abstraction. I refactored the classname to be FhirService instead of FhirServiceRepository since the term "repository" wasn't the best fit. I believe this layer of abstraction provides advantages for us. The first is that if we want to update the FhirClient again in the future for example, the blast radius will be significantly smaller because we only need to make changes to this class. In addition, this removes hard dependencies on the underlying FhirClient.

Just my perspective. Do you think this layer of abstraction is unnecessary? We can pass around the IFhirClient and change this class to extension methods as well

Reply:
Thanks for the context.  I am fine with this as is and we can re-evaluate later if needed based on some of the things we discussed offline.